src/cloud-api-adaptor/versions.yaml

```diff
@@ -23,7 +23,8 @@ tools:
   golang: 1.21.9
   rust: 1.72.0
   protoc: 3.15.0
-  packer: v1.9.4
+  packer: v1.10.0
```

Comment:
We pinned at packer 1.9.4 deliberately as it was the last version before packer switched their license. Do you have to bump this to get it working?

Reply:
From https://developer.hashicorp.com/packer/docs/plugins/install-plugins#plugin-loading-workflow i see starting with v1.10.0 of Packer, we can use packer plugins install with the --path flag to install a plugin from a binary.

If we want to keep packer at same version I can modify code as below please suggest.

```
git clone --depth 1 --single-branch https://github.com/hashicorp/packer-plugin-qemu.git -b ${PACKER_QEMU_VERSION}; \
    cd packer-plugin-qemu; \
    go mod tidy; \
    go build; \
    mkdir -p ~/.config/packer/plugins/github.com/hashicorp/; \
    cp packer-plugin-qemu ~/.config/packer/plugins/github.com/hashicorp/; \
    cd $OLDPWD; \
```

Reply:
Yes, I think that keeping the version is best. Do we understand how/why this failure has started given that the version of packer is pinned?

Reply:
We can't use the latest packer version due to license changes. 1.9.4 is the last one with MPL. So we'll have to stick to it.

Reply:
What if you run `packer init` and also ensure a specific qemu plugin version is used?
Example - https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/azure/image/Makefile#L29
https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/azure/image/ubuntu/azure-ubuntu.pkr.hcl#L4

Reply:
We have added qemu version = "1.1.0" [here](https://github.com/confidential-containers/cloud-api-adaptor/blob/13457162f6c094c67bebd1b2c7b5cde540f2cd33/src/cloud-api-adaptor/podvm/qcow2/rhel/qemu-rhel.pkr.hcl#L3) but then it seems failing as s390x support is missing so added building it from source.

```
packer init qcow2/rhel/qemu-rhel.pkr.hcl
Failed getting the "github.com/hashicorp/qemu" plugin:
21 errors occurred:
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_darwin_amd64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_darwin_arm64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_freebsd_386.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_freebsd_amd64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_freebsd_arm.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_illumos_amd64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_linux_386.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_linux_amd64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_linux_arm.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_linux_arm64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_linux_ppc64le.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_netbsd_386.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_netbsd_amd64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_netbsd_arm.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_openbsd_386.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_openbsd_amd64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_openbsd_arm.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_solaris_amd64.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_windows_386.zip: wrong system, expected linux_s390x 
	* ignoring invalid remote binary packer-plugin-qemu_v1.1.0_x5.0_windows_amd64.zip: wrong system, expected linux_s390x 
	* could not install any compatible version of plugin "github.com/hashicorp/qemu"
```

Reply:
i think ill change code as below, so that i don't need to change packer version.

```
git clone --depth 1 --single-branch https://github.com/hashicorp/packer-plugin-qemu.git -b ${PACKER_QEMU_VERSION}; \
    cd packer-plugin-qemu; \
    go mod tidy; \
    go build; \
    mkdir -p ~/.config/packer/plugins/github.com/hashicorp/; \
    cp packer-plugin-qemu ~/.config/packer/plugins/github.com/hashicorp/; \
    cd $OLDPWD; \
```